metagraph/src/graph/annotated_graph_algorithm.cpp

```diff
@@ -322,37 +322,30 @@ construct_diff_label_count_vector(const AnnotatedDBG &anno_graph,
     const auto &label_encoder = anno_graph.get_annotation().get_label_encoder();
     const auto &binmat = anno_graph.get_annotation().get_matrix();
 
-    std::vector<uint64_t> label_in_codes;
-    label_in_codes.reserve(labels_in.size());
-    std::vector<uint64_t> label_out_codes;
-    label_out_codes.reserve(labels_out.size());
+    std::vector<uint64_t> label_codes;
+    label_codes.reserve(labels_in.size() + labels_out.size());
+    std::vector<bool> is_label_out;
+    is_label_out.reserve(label_codes.size());
     for (const std::string &label_in : labels_in) {
-        label_in_codes.push_back(label_encoder.encode(label_in));
+        label_codes.push_back(label_encoder.encode(label_in));
+        is_label_out.push_back(false);
     }
     for (const std::string &label_out : labels_out) {
-        label_out_codes.push_back(label_encoder.encode(label_out));
+        label_codes.push_back(label_encoder.encode(label_out));
+        is_label_out.push_back(true);
     }
 
     std::mutex vector_backup_mutex;
     std::atomic_thread_fence(std::memory_order_release);
     bool parallel = num_threads > 1;
-    binmat.call_columns(label_in_codes,
-        [&](auto, const bitmap &rows) {
-            rows.call_ones([&](auto r) {
-                node_index i = AnnotatedDBG::anno_to_graph_index(r);
-                set_bit(indicator.data(), i, parallel, MO_RELAXED);
-                atomic_fetch_and_add(counts, i * 2, 1, vector_backup_mutex, MO_RELAXED);
-            });
-        },
-        num_threads
-    );
 
-    binmat.call_columns(label_out_codes,
-        [&](auto, const bitmap &rows) {
+    binmat.call_columns(label_codes,
```

Comment:
As I remember, you do allow having the same label in "IN" and "OUT". So I don't see how this will work properly in that case. Or, am I wrong?

Reply:
I output a warning if it finds duplicates, but I think given this last patch, we should make it write an error and just skip that experiment.

Reply:
Printing a warning is not suitable in this case. If the behavior for a case is unpredictable or wrong, you must not allow that scenario to happen. That is, print an error and exit.
Otherwise, fix this block of code to behave properly.
Theoretically, I see no issues with having a non-zero overlap between IN and OUT. I suppose, there indeed might be practical cases where this will be required.

Reply:
For instance, "find sequences present in at least one of the samples input, but not in all of them".

Reply:
That's a good point, actually. I guess it won't change the code too much to support having overlapping labels.